include/aws/s3/s3_client.h

```diff
@@ -142,6 +142,56 @@ typedef void(aws_s3_meta_request_shutdown_fn)(void *user_data);
 
 typedef void(aws_s3_client_shutdown_complete_callback_fn)(void *user_data);
 
+/**
+ * Information sent in the meta_request telemetry callback.
+ */
+struct aws_s3_request_telemetry_metrics {
+    /* X-AMZ-REQUEST-ID Header value. */
+    struct aws_byte_cursor request_id;
+    /* The IP address of the request connected to */
+    struct aws_byte_cursor ip_address;
+
+    struct {
+        /* The total time for the single request started sending to finish (response received or failed). */
+        uint64_t total_duration_time_ns;
+        /* The time for the request started sending to start receiving response. */
+        uint64_t response_latency_time_ns;
+        /* The time for the request started sending to finish sending. */
+        uint64_t send_time_ns;
+        /* The time for the request started receiving to finish receiving response. */
+        uint64_t receive_time_ns;
+        /* The time stamp when the request started to be sent */
+        uint64_t started_sending_time_stamp;
+    } time_metrics;
+
+    struct {
+        /* Response status code for the request */
+        uint32_t response_status;
+        /* HTTP Headers of the response received. */
+        struct aws_http_headers *response_headers;
+        /* The request line: method, path and query of the request.  */
+        struct aws_byte_cursor request_line;
+    } req_resp_info_metrics;
+
+    struct {
+        /* The pointer ID to the connection that request was made from */
+        size_t connection_id;
```

Comment:
also useful: whether this was the 1st or 40th request on a given connection. maybe named: connection_request_count